File: solutions/go/gigasecond/1/gigasecond.go
// Package gigasecond calculates the moment one gigasecond after a given time.
package gigasecond

import "time"

// AddGigasecond returns the time one gigasecond (10^9 seconds) after t.
func AddGigasecond(t time.Time) time.Time {
	return t.Add(1_000_000_000 * time.Second)
}

File: solutions/python/gigasecond/1/gigasecond.py
"""Calculate the moment one gigasecond after a given moment."""

from datetime import datetime, timedelta


def add(moment: datetime) -> datetime:
    """Return the datetime one gigasecond after the given moment.

    Args:
        moment: The starting datetime.

    Returns:
        A new datetime one gigasecond (1e9 seconds) later.
    """
    gigasecond = timedelta(seconds=1_000_000_000)
    return moment + gigasecond